### PR TITLE
Removes Revolutionaries becoming enemies of the revolution

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

> I haven't been able to test this due to not being able to open 3 instances of BYOND.

This removes the ability for Security/Command Revolutionaries from becoming Enemies of the Revolution upon a Rev victory. Command cannot be revs in any circumstance, so it only affects Security.

I'm marking this as fix for now, but I was told that this might've been intentional. If so, I can swap this out for balance, but it makes no sense why a Revolutionary would become an Enemy of the Revolution just for succeeding at their job.

Tested:
image

## Why It's Good For The Game

step 1) implant removal surgery the warden
step 2) convert them to the revolution
step 3) watch them kill the HoS
step 4) kill them because now they're an enemy of the revolution

## Changelog
:cl:
fix: Security Revolutionaries will no longer become Enemies of the Revolution upon a Revolution victory.
/:cl:

### CODE REVIEW

**Review by Cyberboss — APPROVED** [review]

File: code/modules/antagonists/revolution/revolution.dm [hunk 1]
```diff
@@ -419,6 +419,9 @@
 			if (!(mind.assigned_role in GLOB.command_positions + GLOB.security_positions))
 				continue
 
+			if (!(mind.has_antag_datum(/datum/antagonist/rev))
```
Comment (Cyberboss) [inline code/modules/antagonists/revolution/revolution.dm:422]:
```suggestion
			if (!(mind.has_antag_datum(/datum/antagonist/rev)))
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/antagonists/revolution/revolution.dm [hunk 2]
```diff
@@ -419,6 +419,9 @@
 			if (!(mind.assigned_role in GLOB.command_positions + GLOB.security_positions))
 				continue
 
+			if (!(mind.has_antag_datum(/datum/antagonist/rev)))
+				continue
```
Comment (MrMelbert) [inline code/modules/antagonists/revolution/revolution.dm:423]: If i'm reading this right, this only adds "Enemy of the Revolution" if they ARE a rev, is this tested?
`(if mind does not have a rev antagonist datum, skip this player in the loop.)`

**Comment by MrMelbert** [comment]
I'll also point out (not 100% certain) but I think it removes all rev antag datums by this point in the code so this check might not do anyhting

**Comment by JohnFulpWillard** [comment]
> I'll also point out (not 100% certain) but I think it removes all rev antag datums by this point in the code so this check might not do anything

That was my original fear when I noticed I couldn’t test it locally

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**